client/types/account-overview.d.ts

```diff
@@ -42,11 +42,11 @@ declare namespace AccountOverview {
 
 	interface Overview {
 		currency: string;
-		lastPaid: Deposit;
-		nextScheduled: Deposit;
+		lastPaid?: Deposit;
+		nextScheduled?: Deposit;
 		pending: Balance;
```

Comment:
I still don't quite understand fully why you make lastPaid, nextScheduled, and instant optional and why only those 3? Why didn't you make `pending` and `available` also optional?

Looking at https://github.com/Automattic/woocommerce-payments/blob/fix/5863-account-overview-nextScheduled-typeerror/client/data/deposits/selectors.js#L86-L93
potentially all of them can be optional.

However, the next line https://github.com/Automattic/woocommerce-payments/blob/fix/5863-account-overview-nextScheduled-typeerror/client/data/deposits/selectors.js#L97
is setting each key: `lastPaid`, `nextScheduled`, `pending`, `available`, and `instant`.

Also, looking at server code, the value of `deposit.next_scheduled` will always be defined. At most, it'll be empty array, so I don't understand why we have to handle when it's undefined.

Reply:
I am just rubber ducking. Now I can see why. It's because when it's an empty array, it won't this [forEach](https://github.com/Automattic/woocommerce-payments/blob/fix/5863-account-overview-nextScheduled-typeerror/client/data/deposits/selectors.js#L82) won't even run even once.

Reply:
I also can now see why `pending` and `available` are always defined.

Reply:
As you mentioned, this is where the `undefined` value originates from: `client/data/deposits/selectors.js` `getAllDepositsOverviews()`:

https://github.com/Automattic/woocommerce-payments/blob/b644b98c5c3a83c0969749762d64f7d25c18bda3/client/data/deposits/selectors.js#L86-L93

So I think you're correct that we can safely use `lastPaid: Deposit | undefined;`.

We can also update `pending`, `available` and `instant`.

> Also, looking at server code, the value of deposit.next_scheduled will always be defined.

This `AccountOverview.Overview` type is being used by `AccountOverview.OverviewsResponse.overview.currencies`:

https://github.com/Automattic/woocommerce-payments/blob/b644b98c5c3a83c0969749762d64f7d25c18bda3/client/types/account-overview.d.ts#L52-L58

So the response from the server is being restructured in the `getAllDepositsOverviews()` code above, and the `undefined` value for these properties is being introduced there.

Logging the value of `currencies` from [this line](https://github.com/Automattic/woocommerce-payments/blob/b644b98c5c3a83c0969749762d64f7d25c18bda3/client/data/deposits/selectors.js#L101) confirms that `nextScheduled` can be `undefined` but is present on the currency object:

<img width="831" alt="image" src="https://user-images.githubusercontent.com/3147296/228704329-de2d6b01-40c3-45d6-bbad-1f096351eef5.png">

I'll update the `AccountOverview.Overview` type to ensure all properties can be `undefined`. Since this relates only to the `currencies` array, perhaps we could extend `AccountOverview.Overview` instead?

Reply:
I've updated this in cebf521.

@shendy what do you think of this solution?

Reply:
Looks good @Jinksi.